core/src/main/java/jenkins/util/groovy/GroovyHookScript.java

```diff
@@ -79,10 +83,10 @@ public void run() {
             }
         }
 
-        File script = new File(j.getRootDir(), hookGroovy);
+        File script = new File(home, hookGroovy);
```

Comment:
It is marked `@Nonnull` and I am doing a null check in `BootFailure`. Why do you think it could be null?